pilot/pkg/model/gateway.go

```diff
@@ -27,14 +29,23 @@ type MergedGateway struct {
 
 	// maps from physical port to virtual servers
 	Servers map[uint32][]*networking.Server
+
+	// maps from port names to virtual hosts
```

Comment:
Are we back to port names ? Why ?

Reply:
no we are not. We are simply using the gateway port names to differentiate RDS for each https port

Reply:
'The typical length is 1 for single-SAN SNI certs'. Plaintext and default TLS will typically have a list with all hostnames and have the same length.